mozilla-services/socorro
# fixes bug 1411991 - symbol upload metrics

This implements symbol upload metrics generation.

To test:

1. Run `docker-compose up webapp`
2. Log into the webapp
3. Click on "Symbols" in the bottom right hand corner
4. Upload a symbols file via the web and via the api

You can get symbols files from Tecken because they're awesome. I use this one:

https://github.com/mozilla-services/tecken/blob/master/tests/sample.zip

You can verify metrics were generated by going to `http://localhost:8080` and looking at `statsd.symbols.upload`. Since that's statsd, it won't show tags.

You can also verify metrics were generated by looking at the webapp terminal. You'll see lines like this:

```
webapp_1         | METRICS|2017-11-08 01:50:43|incr|symbols.upload.web_upload|1|#email:wkahngreenemozilla.com
```

We'll have to test the datadog configuration on -stage. It should be very easy to verify yes/no plus verify it's generating tag data we can use in Datadog graphs.

I think this is pretty low risk, but I could be convinced otherwise. The only reason to land it now rather than in a couple of weeks is that we'll be able to accrue symbol upload data and have a better baseline when we migrate.
